src/compiler/scala/tools/nsc/reporters/Reporter.scala

```diff
@@ -150,12 +124,17 @@ abstract class FilteringReporter extends Reporter {
       }
       if (show) {
         positions(fpos) = severity
-        messages(fpos) ::= Reporter.stripExplanation(msg) // ignore explanatory suffix for suppressing duplicates
+        messages(fpos) ::= stripQuickfixable(msg)
       }
       show
     }
   }
 
+  private def stripQuickfixable(msg: String): String = {
+    val i = msg.indexOf(" [quickfixable]")
+    if (i > 0) msg.substring(0, i) else msg
+  }
```

Comment:
I would have `if (msg.endsWith(" [qfa]")) etc`.

Reply:
I noticed it's not always at the end, the "Scala 3 migration messages are errors under -Xsource:3..." appendix is added in RunReporting, after `[quickfixable]`.

Reply:
ok I'm sorry I updated your title.